### PR TITLE
[BEAM-1871] Delete a bunch of deprecated/unused code

### PR DESCRIPTION
**Pull request by dhalperi**
Delete:
* Unused code in IOChannelUtils
* Unused code in Pipeline
* AttemptAndTimeBoundedExponentialBackoff
* AttemptBoundedExponentialBackoff
* IntervalBoundedExponentialBackoff
* AppEngineEnvironment

### CODE REVIEW

**Comment by dhalperi** [comment]
(expecting this to fail until I push a new DataflowWorker that does not depend on some of this deprecated stuff)

**Comment by dhalperi** [comment]
R: @tgroh 

CC: @lukecwik  -- I'll wait for your PR to merge, in case there are conflicts.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java [hunk 1]
```diff
@@ -121,10 +120,7 @@
     public ExecutorService create(PipelineOptions options) {
       ThreadFactoryBuilder threadFactoryBuilder = new ThreadFactoryBuilder();
       threadFactoryBuilder.setThreadFactory(MoreExecutors.platformThreadFactory());
-      if (!AppEngineEnvironment.IS_APP_ENGINE) {
-        // AppEngine doesn't allow modification of threads to be daemon threads.
-        threadFactoryBuilder.setDaemon(true);
-      }
+      threadFactoryBuilder.setDaemon(true);
```
Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: Do we really want to make these daemon threads all the time?

Previously on the Dataflow worker and SDK harness, these were non daemon threads.

The GcsOptions ExecutorService is consumed and used in several places.

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java [hunk 2]
```diff
@@ -133,7 +133,7 @@ public void setup() {
 
   /**
    * Validates that a given time is valid for the total time slept by a
-   * AttemptBoundedExponentialBackOff given the number of retries and
+   * BackOff given the number of retries and
```
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java]: link it?

**Comment by dhalperi** [comment]
Re: Dataflow worker -- not any more; I update the worker in this PR.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java (hunk 1 above)
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: Reading the code, I think this PR is a no-op w.r.t. this issue, assuming we're not in App Engine. Am I missing something?

Comment (lukecwik) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: Your right, my mistake.
Its a noop.

**Comment by dhalperi** [comment]
Btw: PreCommit successfully caught that this would break Dataflow Worker last night: https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9660/org.apache.beam$beam-examples-java/testReport/

But now it passes: https://builds.apache.org/job/beam_PreCommit_Java_MavenInstall/9682/

**Pull request merged**